### PR TITLE
Tutorial updates

### PR DESCRIPTION
**Pull request by IvarStefansson**
## Proposed changes

Overhaul of tutorials. Tutorials are removed, renamed and reworked. The remaining ones should give a decent, up-to-date introduction to PorePy.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Testing (contribution related mainly to testing of existing functionality)
- [x] Documentation update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR._
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation 
- [x] I have checked that I have not duplicated existing functionality

## Further comments

Main updates are related to AD. Note the summary provided in the readme.

### CODE REVIEW

**Review by jhabriel — CHANGES_REQUESTED** [review]
Superb job in updating the tutorials.

I requested some minor changes. After these are addressed, I'll be happy to approve the merging.

Although I don't think this is necessary for this particular PR. I do believe some restructuring of the tutorials is needed. By restructuring, I mean organizing them in separate (sub)directories based on whether they are meant as introductory, technical, advanced, benchmarking (I'm working on these for poroelasticity), or legacy tutorials (which might as well be called deprecated).

File: tutorials/Readme.md [hunk 1]
```diff
@@ -9,8 +9,15 @@ The appropriate order of reading may depend on the reader's background and ambit
...
-5. [ad_framework](./ad_framework.ipynb) describes how to solve a problem using the AD framework. The tutorial includes setup of parameters and discretizations.
-6. [incompressible_flow_model](incompressible_flow_model.ipynb) describes how to use a model class `Incompressible Flow`. The tutorial exposes several extensions and how to solve a mixed-dimensional problem with a few lines of code.
\ No newline at end of file
+2. [grid_structure](./grid_structure.ipynb) describes the structure of individual grids. It demonstrates construction of different types of grids. It also dives into the data structure, and shows how to access and manipulate grid quantities.
+3. [meshing_of_fractures](./meshing_of_fractures.ipynb) describes the construction of mixed-dimensional grids representing a fracture network and the surrounding porous medium.
+4. [algorithmic_differentiation](./algorithmic_differentiation.ipynb) provides an introduction to algorithmic differentiation (AD) and how to solve a generic equation using the AD framework. The tutorial includes setup of parameters and discretizations.
```
[the remark this answers is not in the document]
Comment (IvarStefansson) [reply tutorials/Readme.md:14]: This was @jwboth's suggestion. Could you remind us of the rationale, please?

File: tutorials/flux_discretizations.ipynb [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+   "metadata": {},
+   "source": [
+    "### Multi-point flux approximation "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The multi-point flux approximation also solves the single-phase flow problem in the primal formulation. We discretize the flux term anew with the MPFA, and assemble the matrix. Note that the call to Mpfa.discretize() will override the already stored discretization matrix for Tpfa. We reuse the source term discretization, as this is the same for TPFA and MPFA."
```
[the remark this answers is not in the document]
Comment (IvarStefansson) [reply tutorials/flux_discretizations.ipynb:283]: That could be a solution, yes. Opinions, @keileg?

File: tutorials/Readme.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (keileg) [reply tutorials/Readme.md:14]: I think this depends on the viewpoint: From the user side it is automatic, whereas to a developer it is certainly not so. I have no strong opinions here besides the comment that using algorithmic will confuse some people.

File: tutorials/incompressible_flow_model.ipynb [hunk 3]
```diff
@@ -565,21 +572,42 @@
...
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5ae70b03-3dde-4413-a62d-e3d8586182ce",
+   "metadata": {},
+   "source": [
+    "Unsurprisingly, the resulting pressure solution closely resembles that obtained using the parent class `UnitaryFractureSource`, with an additional high-pressure region around the point $(x,y) = (0.7, 0.7)$.\n",
+    "\n",
+    "## What have we explored\n",
```
[the remark this answers is not in the document]
Comment (keileg) [reply tutorials/incompressible_flow_model.ipynb:600]: I agree.

File: tutorials/flux_discretizations.ipynb (hunk 2 above)
[the remark this answers is not in the document]
Comment (keileg) [reply tutorials/flux_discretizations.ipynb:283]: I think @jhabriel has a valid point that people will be confused. I'm fine with dropping the primal/dual terminology.

File: tutorials/incompressible_flow_model.ipynb (hunk 3 above)
[the remark this answers is not in the document]
Comment (keileg) [reply tutorials/incompressible_flow_model.ipynb:600]: @IvarStefansson (@jhabriel): I can draft such summaries where I find it natural, then you can amend. Is that okay?

File: tutorials/Readme.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jwboth) [reply tutorials/Readme.md:14]: The first sentence of wikipedia on AD: "In mathematics and computer algebra, automatic differentiation (AD), also called algorithmic differentiation, computational differentiation, auto-differentiation, or simply autodiff, is a set of techniques to evaluate the derivative of a function ..." 

Eirik's guess on the viewpoint is I guess the reason for why there exist different names for it. Whenever I heard someone talking about research on AD, they liked using "algorithmic differentiation" to emphasize that there is no free lunch when you think about linearizing more complex problems (see e.g. contact mechanics), while users indeed like to point out that for them it is "automatic".

After all, I also have no strong opinion how to call AD in the doc of porepy.

**Review by keileg — APPROVED** [review]

**Comment by keileg** [comment]
I'm fine with merging when @jhabriel's request has been met.

Thanks a lot for the effort, @IvarStefansson!

**Review by jhabriel — APPROVED** [review]

**Pull request merged**